pages/lessons/fundamentals/wolovim-part-1.mdx

```diff
@@ -0,0 +1,321 @@
+
+<Callout emoji='💡' size='md' variant='info'>
+**Note:** “Ethereum node” and “Ethereum client” are often used interchangeably. In either case, it refers to the software that a participant in the Ethereum network runs. This software can read block data, receive updates when new blocks are added to the chain, broadcast new transactions, and more.
+</Callout>
+
+Ethereum clients can be configured to be reachable by IPC, HTTP, or Websockets. web3.py refers to these connection options as **providers**. You’ll want to choose one of the three providers to configure so that web3.py knows how to communicate with your node.
+
+![https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png](https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png)
+
+The Ethereum node and web3.py need to be configured to communicate via the same protocol, e.g., IPC in this diagram.
+
+Once web3.py is properly configured, you can begin to interact with the blockchain. Here’s a couple of web3.py usage examples as a preview of what’s to come:
+
+```python
+# read block data:
+w3.eth.get_block('latest')
+
+# send a transaction:
+w3.eth.send_transaction({'from': ..., 'to': ..., 'value': ...})
+```
+
+<SideDrawer buttonText="Did you get all that?" title="Comprehension Check">
+      <Question question="eth-intro-1/2-checkin/Q1" />
+      <Question question="eth-intro-1/2-checkin/Q2" />
+      <Question question="eth-intro-1/2-checkin/Q3" />
+      <Question question="eth-intro-1/2-checkin/Q4" />
+      <Question question="eth-intro-1/2-checkin/Q5" />
+      <Question question="eth-intro-1/2-checkin/Q6" />
+</SideDrawer> 
+<br/>
+
+### **Installation**
+
+In this walkthrough, we’ll just be working within a Python interpreter. We won't be creating any directories, files, classes or functions.
+
+<Callout emoji='💡' size='md' variant='info'>
+**Note:** In the examples below, commands that begin with `$` are intended to be run in the terminal. (Do not type the `$`, it just signifies the start of the line.)
+</Callout>
+
+First, install [IPython](https://ipython.org/) for a user-friendly environment to explore in. IPython offers tab completion, among other features, making it much easier to see what’s possible within web3.py.
+
+```bash
+$ pip install ipython
+```
+
+web3.py is published under the name `web3`. Install the latest version like so:
+
+```bash
+$ pip install web3
+```
+
+One more thing – we're going to simulate a blockchain later, which requires a couple more dependencies. You can install those via:
+
+```bash
+$ pip install "web3[tester]"
+```
+
+You’re all set up to go!
+
+### **Spin up a sandbox**
+
+Open up a new Python environment by running `ipython` in your terminal. This is comparable to running `python`, but comes with more bells and whistles.
```

Comment:
maybe say "Open up a new Python environment by running `ipython` in the same ternimal/the terminal you already have open' (something like that - I'm not sure if I should open a new one)